src/DatePicker/DatePicker.tsx

```diff
@@ -109,37 +98,31 @@ const calendarIcon = (
   </svg>
 );
 
-export const Props = {
-  id: t.maybe(t.String),
-  value: t.maybe(t.Date),
-  defaultValue: t.maybe(t.Date),
-  onChange: t.maybe(t.Function),
-  onHide: t.maybe(t.Function),
-  displayFormat: t.maybe(t.String),
-  minDate: t.maybe(t.Date),
-  maxDate: t.maybe(t.Date),
-  fromDate: t.maybe(t.Date),
-  toDate: t.maybe(t.Date),
-  displayTwoMonths: t.maybe(t.Boolean),
-  small: t.maybe(t.Boolean),
-  icon: t.maybe(ReactChild),
-  startDate: t.maybe(t.Date),
-  locale: t.maybe(t.String),
-  onFocusChange: t.maybe(t.Function),
-  disabled: t.maybe(t.Boolean),
-  className: t.maybe(t.String),
-  style: t.maybe(t.Object)
-};
+const clearIcon = (
+  <svg
+    version="1.1"
+    xmlns="http://www.w3.org/2000/svg"
+    height="12"
+    viewBox="0 0 22 28"
+  >
+    <path
+      fill="#9098a7"
+      d="M20.281 20.656q0 0.625-0.438 1.062l-2.125 2.125q-0.438 0.438-1.062 0.438t-1.062-0.438l-4.594-4.594-4.594 4.594q-0.438 0.438-1.062 0.438t-1.062-0.438l-2.125-2.125q-0.438-0.438-0.438-1.062t0.438-1.062l4.594-4.594-4.594-4.594q-0.438-0.438-0.438-1.062t0.438-1.062l2.125-2.125q0.438-0.438 1.062-0.438t1.062 0.438l4.594 4.594 4.594-4.594q0.438-0.438 1.062-0.438t1.062 0.438l2.125 2.125q0.438 0.438 0.438 1.062t-0.438 1.062l-4.594 4.594 4.594 4.594q0.438 0.438 0.438 1.062z"
+    />
+  </svg>
+);
 
 /**
  * A decent and pretty date picker to be used with React
  */
-@props(Props)
 export class DatePicker extends React.PureComponent<DatePicker.Props, State> {
+  dayRenderer: (day: moment.Moment) => JSX.Element;
+
   componentWillMount() {
     if (this.props.locale) {
       moment.locale(this.props.locale);
     }
+    this.dayRenderer = this.dayRendererFactory();
```

Comment:
It's not 100% clear why the indirection `dayRenderer`/`dayRendererFactory` is needed. Maybe we can discuss this offline a few minutes?

Reply:
Discussed offline and now clearer. Just for the sake of completeness, I would add a link to the react-dates issue explaining the need and rationale behind this @veej